src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -895,13 +945,21 @@ private static async Task<DurableOrchestrationStatus> GetDurableOrchestrationSta
                                     ConvertOutputToJToken(historyItem, showHistoryOutput);
                                     break;
                                 case EventType.ExecutionTerminated:
-                                    historyItem.Remove("Input");
+                                    if (!showHistoryInput)
+                                    {
+                                        historyItem.Remove("Input");
+                                    }
+
                                     break;
                                 case EventType.TimerFired:
                                     historyItem.Remove("TimerId");
                                     break;
                                 case EventType.EventRaised:
-                                    historyItem.Remove("Input");
+                                    if (!showHistoryInput)
```

Comment:
Instead of repeating this same code in multiple places, let's just move it down below, outside of the `switch` statement. It's okay if we call `history.Remove("Input")` on a history event that doesn't have an `"Input"` field. It will just be a no-op. This design is safer because we never have to worry about accidentally forgetting any of the `case` statements.